cmd/snap/cmd_set.go

```diff
@@ -155,6 +162,37 @@ func (x *cmdSet) Execute([]string) error {
 	return nil
 }
 
+func validateJsonKeys(v interface{}) error {
+	var initial map[string]interface{}
+
+	switch val := v.(type) {
+	case map[string]interface{}:
+		initial = val
+	case bool, string, json.Number, nil:
+		return nil
+	default:
+		return fmt.Errorf("invalid type: %T", v)
+	}
+
+	stack := []map[string]interface{}{initial}
+
+	for len(stack) > 0 {
+		current := stack[len(stack)-1]
+		stack = stack[:len(stack)-1]
+
+		for key, value := range current {
+			if err := config.ValidateKey(key); err != nil {
+				return err
+			}
+
+			if nestedMap, ok := value.(map[string]interface{}); ok {
+				stack = append(stack, nestedMap)
+			}
```

Comment:
Should this switch on `value.(type)` as above? What if it's an invalid type? Perhaps a shared helper function would be useful, something like `validateJsonValue`, which just does what that switch does and can be used in both places.